### PR TITLE
TSK-2013 selectAndClaim TasK return Optional and does not throw Exception if Task is notFound

### PR DESCRIPTION
**Pull request by arolfes**
[SonarCloud](https://sonarcloud.io/summary/new_code?id=arolfes_taskana&branch=TSK-2013-selectAndClaim-returns-Optional)

---
Release Notes:

```

```
 
### For the submitter:
- [ ] I updated the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) and will supply links to the specific files
- [ ] I did not update the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview)
- [ ] I included a link to the [SonarCloud branch analysis](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1019969636/SonarCloud+Integration)
- [ ] After integration of the PR, I added a description of changes on the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [ ] I did not update the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [ ] I put the ticket in review
- [ ] After integration of the pull request, I verified our [bluemix test environment](http://taskana.mybluemix.net/taskana) is not broken

### Verified by the reviewer:
- [ ] Commit message format → TSK-XXX: Your commit message.
- [ ] Submitter's update to [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) is sufficient
- [ ] SonarCloud analysis meets our standards
- [ ] Update of the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana) reflects changes
- [ ] PR fulfills the ticket
- [ ] Edge cases and unwanted side effects are tested
- [ ] Readability

### CODE REVIEW

File: lib/taskana-core/src/test/java/acceptance/task/claim/SelectAndClaimTaskAccTest.java [hunk 1]
```diff
@@ -56,19 +53,6 @@ void should_ClaimDifferentTasks_For_ConcurrentSelectAndClaimCalls() throws Excep
...
-
-    TaskQuery query = taskanaEngine.getTaskService().createTaskQuery().idIn("notexisting");
-    ThrowingCallable call = () -> taskanaEngine.getTaskService().selectAndClaim(query);
-    assertThatThrownBy(call)
-        .isInstanceOf(SystemException.class)
-        .hasMessageContaining(
-            "No tasks matched the specified filter and sorting options, "
-                + "task query returned nothing!");
-  }
-
```
Comment (mustaphazorgati) [inline lib/taskana-core/src/test/java/acceptance/task/claim/SelectAndClaimTaskAccTest.java:71]: I am expecting a test where Optional.empty() is returned :)

File: lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskServiceImpl.java [hunk 2]
```diff
@@ -584,7 +584,7 @@ public void forceDeleteTask(String taskId)
   }
 
   @Override
-  public Task selectAndClaim(TaskQuery taskQuery)
+  public Optional<Task> selectAndClaim(TaskQuery taskQuery)
```
Comment (mustaphazorgati) [inline lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskServiceImpl.java:587]: What do u think of this?
```
  @Override
  public Optional<Task> selectAndClaim(TaskQuery taskQuery) {
    ((TaskQueryImpl) taskQuery).selectAndClaimEquals(true);
    return taskanaEngine.executeInDatabaseConnection(
        () ->
            Optional.ofNullable(taskQuery.single()).map(TaskSummary::getId).map(wrap(this::claim)));
  }

```
This also removed the 2 thrown exceptions and thus their declaration can be removed aswell

File: rest/taskana-rest-spring/src/main/java/pro/taskana/task/rest/TaskController.java [hunk 3]
```diff
@@ -254,9 +255,11 @@ public ResponseEntity<TaskRepresentationModel> selectAndClaimTask(
     filterCustomIntFields.apply(query);
     sortParameter.apply(query);
 
-    Task selectedAndClaimedTask = taskService.selectAndClaim(query);
+    Optional<Task> selectedAndClaimedTask = taskService.selectAndClaim(query);
 
-    return ResponseEntity.ok(taskRepresentationModelAssembler.toModel(selectedAndClaimedTask));
+    return selectedAndClaimedTask
+        .map(task -> ResponseEntity.ok(taskRepresentationModelAssembler.toModel(task)))
+        .orElseGet(() -> ResponseEntity.notFound().build());
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/main/java/pro/taskana/task/rest/TaskController.java:262]: Please document this behavior in the Javadoc, so that our rest documentation get's updated

**Review by mustaphazorgati — CHANGES_REQUESTED** [review]

File: lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskServiceImpl.java (hunk 2 above)
Comment (arolfes) [reply lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskServiceImpl.java:587]: das funktioniert irgendwie nicht
lass uns gemeinsam drauf schauen

Comment (mustaphazorgati) [reply lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskServiceImpl.java]: fixed

**Review by mustaphazorgati — APPROVED** [review]
